flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/api/java/internal/StreamTableEnvironmentImpl.java

```diff
@@ -0,0 +1,347 @@
+	}
+
+	/**
+	 * Creates an instance of a {@link StreamTableEnvironment}. It uses the {@link StreamExecutionEnvironment} for
+	 * executing queries. This is also the {@link StreamExecutionEnvironment} that will be used when converting
+	 * from/to {@link DataStream}.
+	 *
+	 * @param catalogManager The {@link CatalogManager} to use for storing and looking up {@link Table}s.
+	 * @param tableConfig The configuration of the TableEnvironment.
+	 * @param executionEnvironment The {@link StreamExecutionEnvironment} of the TableEnvironment.
+	 */
+	public static StreamTableEnvironmentImpl create(
+			CatalogManager catalogManager,
+			TableConfig tableConfig,
+			StreamExecutionEnvironment executionEnvironment) {
+		FunctionCatalog functionCatalog = new FunctionCatalog();
+		Executor executor = lookupExecutor(executionEnvironment);
+		Planner planner = PlannerFactory.lookupPlanner(executor, tableConfig, functionCatalog, catalogManager);
+		return new StreamTableEnvironmentImpl(
+			catalogManager,
+			functionCatalog,
+			tableConfig,
+			executionEnvironment,
+			planner,
+			executor
+		);
+	}
+
+	private static Executor lookupExecutor(StreamExecutionEnvironment executionEnvironment) {
+		try {
+			Class<?> clazz = Class.forName("org.apache.flink.table.executor.ExecutorFactory");
+			Method createMethod = clazz.getMethod("create", StreamExecutionEnvironment.class);
+
+			return (Executor) createMethod.invoke(null, executionEnvironment);
+		} catch (Exception e) {
+			throw new TableException(
+				"Could not instantiate the executor. Make sure a planner module is on the classpath",
+				e);
+		}
+	}
+
+	@Override
+	public <T> void registerFunction(String name, TableFunction<T> tableFunction) {
+		TypeInformation<T> typeInfo = UserFunctionsTypeHelper.getReturnTypeOfTableFunction(
+			tableFunction);
+
+		functionCatalog.registerTableFunction(
+			name,
+			tableFunction,
+			typeInfo
+		);
+	}
+
+	@Override
+	public <T, ACC> void registerFunction(String name, AggregateFunction<T, ACC> aggregateFunction) {
+		TypeInformation<T> typeInfo = UserFunctionsTypeHelper.getReturnTypeOfAggregateFunction(
+			aggregateFunction);
+		TypeInformation<ACC> accTypeInfo = UserFunctionsTypeHelper
+			.getAccumulatorTypeOfAggregateFunction(aggregateFunction);
+
+		functionCatalog.registerAggregateFunction(
+			name,
+			aggregateFunction,
+			typeInfo,
+			accTypeInfo
+		);
+	}
+
+	@Override
+	public <T, ACC> void registerFunction(String name, TableAggregateFunction<T, ACC> tableAggregateFunction) {
+		TypeInformation<T> typeInfo = UserFunctionsTypeHelper.getReturnTypeOfAggregateFunction(
+			tableAggregateFunction);
+		TypeInformation<ACC> accTypeInfo = UserFunctionsTypeHelper
+			.getAccumulatorTypeOfAggregateFunction(tableAggregateFunction);
+
+		functionCatalog.registerAggregateFunction(
+			name,
+			tableAggregateFunction,
+			typeInfo,
+			accTypeInfo
+		);
+	}
+
+	@Override
+	public <T> Table fromDataStream(DataStream<T> dataStream) {
```

Comment:
will `fromBoundedStream/toBoundedStream` methods be added in this PR ?

Reply:
Not in this PR.